tests/features/workspace.feature

```diff
@@ -0,0 +1,153 @@
+    Given WorkspaceFactory instance with createWorkspace method 
+    And  Configuration for token 123 that includes service A and B and components 1 and 2
+    When I run createWorkspace method with token 123
+    And  An error with registering a component occurs
+    Then I expect to receive an error
+
+
+Scenario: Call services method returns a map of promises to each service loaded in Workspace
+    Given WorkspaceFactory instance with createWorkspace method
+    And   Configuration for token 123 that includes service A and B and components 1 and 2
+    And   Service A and service B include a bootstrap that calls registerService
+    And   the bootstrap includes CAPSULAHUB_WORKSPACE and CAPSULAHUB_CONFIGURATION variable
+    When  I run createWorkspace method with token 123 and Workspace is created
+    And   I call services method
+    Then  I expect to receive a map of promises to service A and B having the following <property>s
+          |<property> |
+          |serviceName|
+          |displayName|
+          |proxy      |
+    And   each of the promises is resolved with corresponding service 
+
+Scenario: Call components method returns a map of promises to each component loaded in Workspace
+    Given WorkspaceFactory instance with createWorkspace method
+    And   Configuration for token 123 that includes service A and B and components 1 and 2
+    When  I run createWorkspace method with token 123 and Workspace is created
+    And   I call workspace components method
+    Then  I expect to receive a map of promises to component 1 and 2 with the following <property>s
+          |<property>   |
+          |componentName|
+          |path         |
+          |config       |
+
+Scenario: Call registerService method registers the provided service in the Workspace
+    Given WorkspaceFactory instance with createWorkspace method
+    And  Configuration for token 123 that includes service A and B and components 1 and 2
+    And  Service A and service B include a bootstrap that call registerService
+    When I run createWorkspace method with token 123 and Workspace is created
+    And  I call registerService method with service A and with a valid request 
+	And  The request has the following <property> and <type>
+         |<property> | <type>|
+         |serviceName| string|
+         |displayName| string|
+         |reference  | any   |
+    Then Service A is registered in the Workspace
+
+Scenario: Call registerService method with a service already registered is rejected with error
+    Given WorkspaceFactory instance with createWorkspace method
+    And   Configuration for token 123 that includes service A and B and components 1 and 2
+    And   Service A and service B include a bootstrap that calls registerService
+    And   I run createWorkspace with token 123 and Workspace is created
+    And   I call registerService method with service A and service is registered
```

Comment:
Again if it's included in the bootstrap of service A so `registerService` will be called from there already